docs/How-To-Run-Query-on-IDP-DB-In-Ruby-Console.rst

```diff
@@ -0,0 +1,98 @@
+.. _how-to-run-query-on-idp-db-in-ruby-console:
+
+How To Run a Custom Query on IDP DB in Ruby Console
+==============================================================================
+
+.. contents::
+    :local:
+
+
+1. Get the Utility Shell Scripts from ``devops`` Repo
+------------------------------------------------------------------------------
+
+- Clone the repo: https://github.com/18F/identity-devops
+- Setup your ruby following this guide: https://github.com/18F/identity-idp. Finally you should able to see your ruby version is 2.3.X by running ``ruby -v`` command.
+- Setup your PIV card auth, follow this guide: https://github.com/18F/identity-private/wiki/Operations:-MacOSX-PIV-to-SSH-key-extraction. This should be already be done as part of your on-boarding process.
+
+
+2. SSH to prod instance, log in to Ruby Console
+------------------------------------------------------------------------------
+
+1. Establish SSH to current IDP prod server (ec2 instance):
+
+first:
+
+.. code-block:: bash
+
+    cd <path-to-identity-devops>
+
+stay in root dir, and:
+
+.. code-block:: bash
+
+    ./bin/ssh-instance --newest asg-prod-idp
+
+2. Once you have access to the IDP server, then get in ruby console:
+
+first:
+
+.. code-block:: bash
+
+    cd /srv/idp/current
+
+then:
+
+.. code-block:: bash
+
+    sudo -uwebsrv RAILS_ENV=production ALLOW_CONSOLE_DB_WRITE_ACCESS=true bundle exec rails c
+
+or:
+
+.. code-block:: bash
+
+    sudo -uwebsrv RAILS_ENV=production bundle exec rails c
+
+.. warning::
+
+    By default, the ruby console uses a read-only account. However, if the goal is to execute write operations, you can set  ``ALLOW_CONSOLE_DB_WRITE_ACCESS=true`` to switch to write mode, assuming your account has the required privileges and role.
```

Comment:
rephrased this a bit, see if it makes sense. i think your account also needs to have write permissions before you can execute a write operation, simply setting ``ALLOW_CONSOLE_DB_WRITE_ACCESS=true`` is not sufficient, correct?

Reply:
There are only 2 db users, the app user and the console user. Setting `ALLOW_CONSOLE_DB_WRITE_ACCESS` tells the console to use the app user instead of the console user. There is no need to worry about permissions for the user that is signed into the IDP.

Reply:
That's to say, if you can log onto an IDP and get sudo privileges, setting `ALLOW_CONSOLE_DB_WRITE_ACCESS` is sufficient.